lib/krane/template_sets.rb

```diff
@@ -39,28 +40,44 @@ def ejson_secrets_file
 
       def validate
         errors = []
+        supported_extensions = if @render_erb
+          TemplateSets::VALID_TEMPLATES
+        else
+          TemplateSets::VALID_TEMPLATES.reject { |extension| extension.include?('erb') }
+        end
+
         if Dir.entries(@template_dir).none? do |filename|
-             filename.end_with?(*TemplateSets::VALID_TEMPLATES) ||
+             filename.end_with?(*supported_extensions) ||
              filename.end_with?(EjsonSecretProvisioner::EJSON_SECRETS_FILE)
            end
-          return errors << "Template directory #{@template_dir} does not contain any valid templates"
+          return errors << "Template directory #{@template_dir} does not contain any valid templates (supported " \
+              "suffixes: #{supported_extensions.join(', ')}, or #{EjsonSecretProvisioner::EJSON_SECRETS_FILE})"
+        end
+
+        if !@render_erb && deploying_with_erb_files?
+          errors << "ERB template discovered with rendering disabled. If you were trying to render ERB and " \
+            "deploy the result, try piping the output of `krane render` to `krane-deploy` with the --stdin flag"
```

Comment:
To truly only raise this message once, we need to move the check out of `TemplateSet#validate` and into `TemplateSets#validate`, no?

Reply:
Pushed up what I think is a fix. Have a look please!